File: CodingSamples/DotNet/Communication/TcpClientTest/DemoApp/ItemInfo.cs
readonly record struct ItemInfo(double Price, int Stock)
{
    public static ItemInfo Parse(string info) //info="price=8000,stock=42"
    {
        string[] parts = info.Split(',');
        double price = double.Parse(parts[0].Substring(6));
        int stock = int.Parse(parts[1].Substring(6));
        return new ItemInfo(price, stock);
    }
}

File: CodingSamples/Java/Communication/HttpClientTest/ItemInfo.java
record ItemInfo(double price, int stock){

	//text="price=19000,stock=100"
	public static ItemInfo parse(String text) {
		String[] parts = text.split(",");
		double p = Double.parseDouble(parts[0].substring(6));
		int s = Integer.parseInt(parts[1].substring(6));
		return new ItemInfo(p, s);
	}
}

